torchvision/datasets/voc.py

```diff
@@ -0,0 +1,270 @@
+    """`Pascal VOC <http://host.robots.ox.ac.uk/pascal/VOC/>`_ Detection Dataset.
+
+    Args:
+        root (string): Root directory of the VOC Dataset.
+        year (string, optional): The dataset year, supports years 2007 to 2012.
+        image_set (string, optional): Select the image_set to use, ``train``, ``trainval`` or ``val``
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+        class_to_ind (dict, optional): dictionary lookup of classnames -> indexes
+            (default: alphabetic indexing of VOC's 20 classes).
+        keep_difficult (boolean, optional): keep difficult instances or not.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+    """
+
+    def __init__(self,
+                 root,
+                 year='2012',
+                 image_set='train',
+                 download=False,
+                 class_to_ind=None,
+                 keep_difficult=False,
+                 transform=None,
+                 target_transform=None):
+        self.root = root
+        self.year = year
+        self.url = DATASET_YEAR_DICT[year]['url']
+        self.filename = DATASET_YEAR_DICT[year]['filename']
+        self.md5 = DATASET_YEAR_DICT[year]['md5']
+        self.transform = transform
+        self.target_transform = target_transform
+        self.image_set = image_set
+        self.class_to_ind = class_to_ind or dict(
+            zip(VOC_CLASSES, range(len(VOC_CLASSES))))
+        self.keep_difficult = keep_difficult
+        _base_dir = DATASET_YEAR_DICT[year]['base_dir']
+        _voc_root = os.path.join(self.root, _base_dir)
+        _image_dir = os.path.join(_voc_root, 'JPEGImages')
+        _annotation_dir = os.path.join(_voc_root, 'Annotations')
+
+        if download:
+            download_extract(self.url, self.root, self.filename, self.md5)
+
+        if not os.path.isdir(_voc_root):
+            raise RuntimeError('Dataset not found or corrupted.' +
+                               ' You can use download=True to download it')
+
+        _splits_dir = os.path.join(_voc_root, 'ImageSets/Main')
+
+        _split_f = os.path.join(_splits_dir, image_set.rstrip('\n') + '.txt')
+
+        if not os.path.exists(_split_f):
+            raise ValueError(
+                'Wrong image_set entered! Please use image_set="train" '
+                'or image_set="trainval" or image_set="val" or a valid'
+                'image_set from the VOC ImageSets/Main folder.')
+
+        self.images = []
+        self.annotations = []
+        with open(os.path.join(_split_f), "r") as lines:
+            for line in lines:
```

Comment:
Can't we instead do
```python
with open(os.path.join(_split_f), "r") as f:
    image_names = f.readlines()
```
? I believe this strips out the `\n` in the end, and is a bit faster than the current version.

Reply:
I've checked and it doesn't strip the `\n`
the value of image_names is:
```
['2008_000008\n', '2008_000015\n', '2008_000019\n', '2008_000023\n', '2008_000028\n', '2008_000033\n', '2008_000036\n', '2008_000037\n', '2008_000041\n', '2008_000045\n', '2008_000053\n', '2008_000060\n', '2008_000066\n', '2008_000070\n', '2008_000074\n', '2008_000085\n', '2008_000089\n', '2008_000093\n', '2008_000095\n', '2008_000096\n', '2008_000097\n', '2008_000099\n', '2008_000103\n', '2008_000105\n', '2008_000109\n', '2008_000112\n'...
```

Reply:
ok. I think what I had done then was something like
```
with open(os.path.join(_split_f), "r") as f:
    image_names = [x.strip() for x in f.readlines()]
```
but do it as you think it's better.